File: kotway/models/events/client_events/clientpageevent.py
from pydantic import BaseModel
from enum import Enum

class ClientPageEventName (Enum):
    UNHANDLED_ROUTE = "unhandled_route"

class ClientPageEvent (BaseModel):
    event_name: ClientPageEventName
    data: dict


class ClientPageEventUnhandeldRoute (BaseModel):
    route: str

File: kotway/web/js/models/clientpageevent.js


export const ClientPageEventName = Object.freeze({
    UNHANDLED_ROUTE: "unhandled_route"
});

export class ClientPageEvent {
    constructor (data = {}) {
        this.event_name = data.event_name;
        this.data = data.data;
    }
}

export class ClientPageEventUnhandeldRoute {
    constructor (data = {}) {
        this.route = data.route;
    }
}